### DISCUSSION TITLE
loco rs support htmx

### DISCUSSION THREAD

**Discussion by ghost** [loco-rs/loco#485] [Q&A]
loco rs support htmx ???

**Comment by schungx** [comment]
Yes it does.  As a matter of fact, I'm developing an entire LOB app with HTMX and Loco, trying to do without a single line of JavaScript.  So far so good.

But of course, since HTMX works with HTML instead of JSON, the examples are not gonna help you.  Your main public workhorse will be a bunch of HTML API's.

Technically you don't need all those JSON API's (unless you need them for other purposes).  However, I find that it is best to expose them anyway, because you're gonna need them anyway -- Loco uses Tera as a templating engine, which accepts JSON inputs.  Therefore, you'll find that you're calling your JSON API in your HTML request handler, then rendering HTML for return.

**Comment by adrian5** [comment]
@i1kaaasra This belongs in **Q&A**, not **Announcements**!

**Comment by azambuilds** [comment]
Is there an example of using HTMX with a templating engine like Tera? Like an example project out there? Anything would do but I'm having a tough time understanding exactly what the SSR templating engine should be giving to htmx, and what kind of static files should be served

**Comment by schungx** [comment]
There are a few things that I hate in Tera when working with HTMX:

* lack of ability to define an object literal (it works well with objects, but cannot _define_ one...  how lazy can the author be?)...

* its very limited expression parser, always requiring defining intermediate variables...  so lots and _lots_ of variables...  and lots and _lots_ of parameters on some macros...

* lack of ability to directly render/call a macro from Rust (I'd say this is a glaring oversight of the API)...

* lack of a good syntax highlighter/formatter for VScode...

However, I have since developed countermeasures to most of them...  although I haven't written them up.  Ping me if you need details, or have questions.
